controller/Controller.php

```diff
@@ -812,7 +919,29 @@ function CheckoutBasket() {
     $basket = CreateSafeOrder($Order->getAllOrdersByOrderStatusNameAndCustomerID("basket", $_SESSION["uid"])[0]);
     if (is_null($basket)) return false;
 
-    return $Order->updateOrderDetails($basket->getOrderID(), "OrderStatusID", $Order->getOrderStatusIDByName("ready"));
+    $orderID = $basket->getOrderID();
+    $basket = GetOrderByID($orderID);
+    // For each orderLine, reduce the stock of the product by the quantity in the orderLine
+    $orderLines = $basket -> getOrderLines();
+    foreach ($orderLines as $orderLine) {
+        $productID = $orderLine -> getProductID();
+        $quantity = $orderLine -> getQuantity();
+        $product = GetProductByID($productID);
+        $updatedStock = $product -> getStock() - $quantity;
+        if ($product -> getStock() < $quantity) {
+            return false;
+        }
+        $result = UpdateProductDetail($productID, "Stock", $updatedStock);
+        echo $result;
```

Comment:
do we want to let the customer see this error or was this just for debugging?

Reply:
Debugging will remove this in a commit